windows/CodePushPackage.cs

```diff
@@ -0,0 +1,364 @@
+            }
+            catch (FileNotFoundException)
+            {
+                return null;
+            }
+        }
+
+        public async Task<string> GetCurrentPackageHash()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            string currentPackageShortHash = (string)info[CURRENT_PACKAGE_KEY];
+            if (currentPackageShortHash == null)
+            {
+                return null;
+            }
+
+            JObject currentPackageMetadata = await GetPackage(currentPackageShortHash);
+            if (currentPackageMetadata == null)
+            {
+                return null;
+            }
+            else
+            {
+                return (string)currentPackageMetadata[PACKAGE_HASH_KEY];
+            }
+        }
+
+        public async Task<string> GetPreviousPackageHash()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            string previousPackageShortHash = (string)info[PREVIOUS_PACKAGE_KEY];
+            if (previousPackageShortHash == null)
+            {
+                return null;
+            }
+
+            JObject previousPackageMetadata = await GetPackage(previousPackageShortHash);
+            if (previousPackageMetadata == null)
+            {
+                return null;
+            }
+            else
+            {
+                return (string)previousPackageMetadata[PACKAGE_HASH_KEY];
+            }
+        }
+
+        public async Task<JObject> GetCurrentPackage()
+        {
+            string packageHash = await GetCurrentPackageHash();
+            if (packageHash == null)
+            {
+                return null;
+            }
+
+            return await GetPackage(packageHash);
+        }
+        
+        public async Task<JObject> GetPreviousPackage()
+        {
+            string packageHash = await GetPreviousPackageHash();
+            if (packageHash == null)
+            {
+                return null;
+            }
+
+            return await GetPackage(packageHash);
+        }
+
+        public async Task<JObject> GetPackage(string packageHash)
+        {
+            StorageFolder packageFolder = await GetPackageFolder(packageHash, false);
+            if (packageFolder == null)
+            {
+                return null;
+            }
+
+            try
+            {
+                StorageFile packageFile = await packageFolder.GetFileAsync(PACKAGE_FILE_NAME);
+                return await CodePushUtils.GetJObjectFromFile(packageFile);
+            }
+            catch (IOException)
+            {
+                return null;
+            }
+        }
+
+        public string shortenPackageHash(string packageHash)
+        {
+            return packageHash.Substring(0, 8);
+        }
+
+        public async Task DownloadPackage(JObject updatePackage, string expectedBundleFileName, Progress<HttpProgress> downloadProgress)
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
```

Comment:
I would recommend adding comments above each of those 4 line chunks of code.

e.g. `// Get the folder of the new update using its hash`